tachyon/math/finite_fields/cubic_extension_field.h

```diff
@@ -78,6 +78,12 @@ class CubicExtensionField : public CyclotomicMultiplicativeSubgroup<Derived> {
     return 3 * BaseField::ExtensionDegree();
   }
 
+  constexpr static void GetRootOfUnity(size_t log_n, Derived& a) {
+    for (size_t i = 0; i < log_n; ++i) {
+      a.SquareInPlace();
+    }
+  }
```

Comment:
Could you show me the reference?

Reply:
[Here](https://github.com/Plonky3/Plonky3/blob/d9ef3902abcc6a34b75f6820da168460aee26763/commit/src/domain.rs#L135)! I wanted to use a similar `GetRootOfUnity()` style that I used and implemented for extension fields for cases like [here](https://github.com/Plonky3/Plonky3/blob/d9ef3902abcc6a34b75f6820da168460aee26763/commit/src/domain.rs#L98) and [here](https://github.com/Plonky3/Plonky3/blob/d9ef3902abcc6a34b75f6820da168460aee26763/commit/src/domain.rs#L120), since it could technically be implemented using `GetSuccessivePowers()` or `GetRootsOfUnity`, but `GetRootOfUnity()` itself in `prime_field_base.h` appears to not to follow the same logic of using a parameter as the generator.